courier/sms.go

```diff
@@ -0,0 +1,112 @@
+package courier
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"net/http"
+
+	"github.com/gofrs/uuid"
+
+	"github.com/ory/kratos/request"
+)
+
+type sendSMSRequestBody struct {
+	To   string
+	From string
+	Body string
+}
+
+type smsClient struct {
+	*http.Client
+	RequestConfig json.RawMessage
+
+	GetTemplateType        func(t SMSTemplate) (TemplateType, error)
+	NewTemplateFromMessage func(d Dependencies, msg Message) (SMSTemplate, error)
+}
+
+func newSMS(ctx context.Context, deps Dependencies) *smsClient {
+	if !deps.CourierConfig(ctx).CourierSMSEnabled() {
+		deps.Logger().Error("messages will not be sent - no sms gate server address is set in config")
+	}
+
+	return &smsClient{
+		Client:        &http.Client{},
```

Comment:
Please use `retryablehttp.Client` with the default HTTP client from `deps` and add a test that e.g. `localhost` can not be used if the security config flag `clients.http.disallow_private_ip_ranges` is `true`.